### PR TITLE
tests: kernel: profiling: Fix _sys_soc_suspend logic

### PR DESCRIPTION
**Pull request by ramakrishnapallala**
Call k_thread_foreach only once from _sys_soc_suspend()
otherwise it will flood the console with stack dumps. As per
the test implementation, the one time call happens through
test_call_stacks_analyze_idle test case.

Signed-off-by: Ramakrishna Pallala <ramakrishna.pallala@intel.com>

### CODE REVIEW

**Comment by vonhust** [comment]
I think it's ok to only check thread->stack_info.start.  For stack size, the real size may vary with different arch.

**Review by rgundi — APPROVED** [review]
LGTM

**Comment by nashif** [comment]
@ramakrishnapallala this does not fix the issue

File: tests/kernel/profiling/profiling_api/src/main.c [hunk 1]
```diff
@@ -88,9 +97,10 @@ static void thread_callback(const struct k_thread *thread, void *user_data)
 		thread_flag = true;
 	}
 
+
 	if (((char *)thread->stack_info.start ==
-			K_THREAD_STACK_BUFFER(test_stack)) &&
-			(thread->stack_info.size == STACKSIZE)) {
+			K_THREAD_STACK_BUFFER(test_stack))
+			&& thread->stack_info.size) {
```
Comment (andrewboie) [inline tests/kernel/profiling/profiling_api/src/main.c]: You could probably just drop the check of stack_info.size all together.

**Review by tyrantlizard77 — APPROVED** [review]

File: tests/kernel/profiling/profiling_api/src/main.c (hunk 1 above)
Comment (ramakrishnapallala) [reply tests/kernel/profiling/profiling_api/src/main.c]: Ok will do.

**Review by nashif — APPROVED** [review]

**Pull request merged**